hitbox-redis/src/actor.rs

```diff
@@ -3,89 +3,303 @@ use crate::error::Error;
+/// [RedisBackend]: RedisBackend
+#[cfg(feature = "single")]
+pub struct RedisSingleBackend {
+    master: ConnectionManager,
+    slave: ConnectionManager,
+}
+
+#[cfg(feature = "single")]
+impl RedisSingleBackend {
+    /// Create new backend instance for redis single instance with default settings.
+    async fn new() -> Result<Self, Error> {
         Self::builder().build().await
     }
 
-    /// Creates new RedisBackend builder with default settings.
-    pub fn builder() -> RedisBackendBuilder {
-        RedisBackendBuilder::default()
+    /// Creates new RedisSingleBackend builder with default settings.
+    fn builder() -> RedisSingleBuilder {
+        RedisSingleBuilder::default()
+    }
+
+    /// Creates new RedisBackend from RedisSingleBackend
+    pub fn finish(self) -> RedisBackend<Self> {
+        RedisBackend { backend: self }
+    }
+}
+
+#[cfg(feature = "single")]
+impl RedisConnection for RedisSingleBackend {
+    type Connection = ConnectionManager;
+
+    fn get_master_connection(&self) -> Self::Connection {
+        self.master.clone()
+    }
+
+    fn get_slave_connection(&self) -> Self::Connection {
+        self.slave.clone()
     }
 }
 
-/// Part of builder pattern implemetation for RedisBackend actor.
-pub struct RedisBackendBuilder {
+/// Part of builder pattern implemetation for RedisBackend actor in single instance.
+#[cfg(feature = "single")]
+pub struct RedisSingleBuilder {
     connection_info: String,
 }
 
-impl Default for RedisBackendBuilder {
+#[cfg(feature = "single")]
+impl Default for RedisSingleBuilder {
     fn default() -> Self {
         Self {
-            connection_info: "redis://127.0.0.1/".to_owned(),
+            connection_info: "redis://127.0.0.1:6379".to_owned(),
         }
     }
 }
 
-impl RedisBackendBuilder {
+#[cfg(feature = "single")]
+impl RedisSingleBuilder {
     /// Set connection info (host, port, database, etc.) for RedisBackend actor.
     pub fn server(mut self, connection_info: String) -> Self {
         self.connection_info = connection_info;
         self
     }
 
     /// Create new instance of Redis backend with passed settings.
-    pub async fn build(&self) -> Result<RedisBackend, Error> {
+    pub async fn build(&self) -> Result<RedisSingleBackend, Error> {
         let client = Client::open(self.connection_info.as_str())?;
         let connection = client.get_tokio_connection_manager().await?;
-        Ok(RedisBackend { connection })
+        Ok(RedisSingleBackend {
+            master: connection.clone(),
+            slave: connection,
+        })
+    }
+}
+
+/// Redis cluster backend based on redis_cluster_async crate.
+/// redis_cluster_async based on redis-rs and gives similar functionality
+/// for working with reids.
+///
+/// This backend create connections to redis cluster instance and
+/// create actor [RedisBackend], which provides redis as storage [Backend] for hitbox.
+/// Its use one [MultiplexedConnection] for asynchronous network interaction.
```

Comment:
Maybe it use multiple Multiplexed connections?

Reply:
Yup, my mistake